002_simulation_structure/002_simulation_structure.tex

```diff
@@ -105,4 +105,63 @@ \subsection{Cost of Living: \texorpdfstring{\lstinline$hpDecay()$}{hpDecay}}\lab
     \text{newHP} = \text{HPCritical}
 \end{equation}
 
-with the \texttt{daysAtCritical} counter incremented by 1. If \texttt{daysAtCritical} reaches \texttt{MaxDayCritical}, the agent dies and is replaced. This counter is reset to 0 if an agent exits the critical state.
\ No newline at end of file
+with the \texttt{daysAtCritical} counter incremented by 1. If \texttt{daysAtCritical} reaches \texttt{MaxDayCritical}, the agent dies and is replaced. This counter is reset to 0 if an agent exits the critical state.
+
+
+
+
+%%%%%%%%
+%%%%%%%% Global Utility
+%%%%%%%%
+\section{Global Utility}\label{global_utility}
+
+To assess the performance of the agents in the tower as a group, we need to define a metric. More insightful than the number of deaths, the notion of \emph{global utility} is of great help in this context [Add reference]. For this project, we implement the definition of global utility used in [Add reference].
+
+Our system is composed of $N$ agents that can perform specific actions in relation with the common pool resources. In a general context, each agent 
+$i\in\{1, \ldots, N\}$ takes the following actions at each iteration $t\in\{1,\ldots,\infty\}$:
+
+\begin{enumerate}
+    \item Determines the resources it has available, $g_i \in [0,1]$.
+    \item Determines its need for resources, $q_i \in [0,1]$.
+    \item Makes a provision of resources, $p_i \in [0,1]$.
+    \item Makes a demand for resources, $d_i \in [0,1]$.
+    \item Receives an allocation of resources, $r_i \in [0,1]$.
+    \item Makes an appropriation of resources, $r'_i \in [0,1]$.
+\end{enumerate}
+
+In the current setup, the available resources $q_i$ corresponds to the current amount of food on the platform.  The need for resources $q_i$ is defined in relation with the health of the agents. We set the following values to $q_i$:
+
+\begin{equation}\label{resources_needed}
+    q_i=\begin{cases}
+     \frac{\mathit{numberDaysInCriticalState}}{\mathit{maxDaysInCriticalState}} & \mbox{if } \mathit{currentHP\leq \mathit{weakLevel}}  \\ 
+     0 & \mbox{else.}
+     \end{cases}
+\end{equation}
+
+This way, we ensure that $q_i$ is bounded by 1 and is proportional to the days spent in the critical health zone below \textit{weakLevel}.
+
+Moreover, the agents do not make any provision $p_i$ to the common pool, as they cannot give food to the platform ($p_i=0$). Their demands for resources is the food they ask when the platform is at their level. In addition, the agents appropriate all resources they are allocated, so that $r'_i=r_i$.\footnote{We ensure that all these parameters are effectively in the range $[0,1]$ by dividing the mentioned quantities by their maximum values.}
```

Comment:
```suggestion
Moreover, the agents do not make any provision $p_i$ to the common pool, as they cannot give food to the platform ($p_i=0$). Their demands for resources is the food they ask for when the platform is at their level. In addition, the agents appropriate all resources they are allocated, so that $r'_i=r_i$.\footnote{We ensure that all these parameters are constrained to the range $[0,1]$ by dividing the mentioned quantities by their maximum values.}
```